### PR TITLE
REF: share code ArrayManager/BlockManager

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -247,16 +245,6 @@ def __nonzero__(self) -> bool:
     # Python3 compat
     __bool__ = __nonzero__
 
-    @property
-    def shape(self) -> Shape:
-        return tuple(len(ax) for ax in self.axes)
-
-    def _normalize_axis(self, axis):
-        # switch axis to follow BlockManager logic
```
Comment (jorisvandenbossche) [inline pandas/core/internals/managers.py:255]: Can you leave this one in their respective classes? 
Although the code is the same, they actually have the opposite purpose (they are used in different places in AM and BM, which probably points to the fact I should have named them differently ..), and while converting more methods to normal axis logic (xref https://github.com/pandas-dev/pandas/pull/40075), the method will also change in AM code.

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
Thanks!

Just one comment about not sharing `_normalize_axis`

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:255]: sure, will update

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -247,12 +245,8 @@ def __nonzero__(self) -> bool:
...
+    def _normalize_axis(self, axis: int) -> int:
+        # switch axis
```
Comment (jorisvandenbossche) [inline pandas/core/internals/managers.py:249]:
```suggestion
        # switch axis to follow BlockManager logic
```

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**